runtime/src/chpl-visual-debug.c

```diff
@@ -0,0 +1,357 @@
+/*
+ * Copyright 2015 Cray Inc.
+ * Other additional copyright holders may be indicated within.
+ * 
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * 
+ * You may obtain a copy of the License at
+ * 
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+//
+// Visual Debug Support file
+//
+
+#include "chpl-visual-debug.h"
+#include "chplrt.h"
+#include "chpl-comm.h"
+#include <stdint.h>
+#include <string.h>
+#include <unistd.h>
+#include <stdio.h>
+#include <stdarg.h>
+#include <fcntl.h>
+#include <errno.h>
+#include <sys/stat.h>
+
+int chpl_vdebug_fd = -1;
+int chpl_vdebug = 0;
+
+int chpl_dprintf (int fd, const char * format, ...)
+{
+  char buffer[2048];
+  va_list ap;
+  int wrv;
+  int retval;
+
+  va_start (ap, format);
+  retval = vsnprintf (buffer, 2048, format, ap);
```

Comment:
If you used sizeof(buffer) here instead of 2048 you would avoid the potential problem of someone down the road changing the buffer size without updating the vnsprintf() limit.